code/obj/item.dm

```diff
@@ -382,77 +382,200 @@
+			return
 
-		playsound(M.loc,"sound/items/eatfood.ogg", rand(10, 50), 1)
-		eat_twitch(M)
-		SPAWN_DBG(1 SECOND)
-			if (!src || !M || !user)
-				return
-			M.visible_message("<span class='alert'>[M] finishes eating [src].</span>",\
-			"<span class='alert'>You finish eating [src].</span>")
-			SEND_SIGNAL(M, COMSIG_ITEM_CONSUMED, user, src)
-			user.u_equip(src)
-			qdel(src)
-		return 1
+		if(M_is_user && !is_it_organs)
+			bar.icon = null // Action bars
+			border.icon = null // minus action bar
+		else
+			eat_twitch(M)
 
-	else
-		user.tri_message("<span class='alert'><b>[user]</b> tries to feed [M] [src]!</span>",\
-		user, "<span class='alert'>You try to feed [M] [src]!</span>",\
-		M, "<span class='alert'><b>[user]</b> tries to feed you [src]!</span>")
-		logTheThing("combat", user, M, "attempts to feed [constructTarget(M,"combat")] [src] [log_reagents(src)]")
+		M.on_eat(master)
 
-		if (!do_mob(user, M))
-			return 0
-		if (get_dist(user,M) > 1)
-			return 0
+		if (src.M_is_user)
+			if(is_it_organs)
+				M.visible_message("<span class='alert'><b>[M]</b> starts cramming \the [master] into [his_or_her(M)] mouth[prob(30) ? " like a [pick(src.grody_adj)] [pick(src.grody_noun)]" : ""]!</span>",\
+				"<span class='[is_awful_monsterthing ? "notice" : "alert"]'>You start cramming \the [master] into your mouth!</span>")
+			else
+				boutput(M, "<span class='notice'>You go to take a bite out of [master].</span>")
+		else
+			user.tri_message("<span class='alert'><b>[user]</b> tries to feed [M] [master]!</span>",\
+			user, "<span class='alert'>You try to feed [M] [master]!</span>",\
+			M, "<span class='[is_awful_monsterthing ? "notice" : "alert"]'><b>[user]</b> tries to feed you [master]!</span>")
+		logTheThing("combat", user, M, "attempts to feed [constructTarget(M,"combat")] [master] [log_reagents(master)]")
+
+	onInterrupt(flag)
+		. = ..()
+		if(src.M_is_user)
+			if(src.is_it_organs)
+				M.visible_message("<span class='alert'>[M] spits out \the [master].</span>","<span class='[is_awful_monsterthing ? "alert" : "notice"]'>You spit out \the [master].</span>")
+			else
+				boutput(M, "<span class='notice'>You stop trying to eat [master].</span>")
+		else
+			if(src.is_it_organs)
+				user.tri_message("<span class='alert'><b>[user]</b> stops forcing \the [master] down [M]'s throat!</span>",\
+				user, "<span class='alert'>You remove \the [master] from [M]'s face!</span>",\
+				M, "<span class='[is_awful_monsterthing ? "alert" : "notice"]'><b>[user]</b> stops forcing \the [master] down your throat!</span>")
+			else
+				user.tri_message("<span class='alert'><b>[user]</b> stops trying to shove \the [master] down [M]'s throat!</span>",\
+				user, "<span class='alert'>You remove \the [master] from [M]'s face!</span>",\
+				M, "<span class='alert'><b>[user]</b> stops trying to shove \the [master] down your throat!</span>")
+
+	onUpdate()
+		..()
+		if(src.failchecks())
+			interrupt(INTERRUPT_ALWAYS)
+			return
```

Comment:
useless

Reply:
How so? I noticed another actionbar having all the interrupt checks in its own proc, and I thought it'd be cleaner to do that instead of repeating the same lines.

Reply:
the return. i always comment on the applicable lines only unless i say otherwise

Reply:
Ohh I getcha, thought there needed to be a return there. Guess not, thanks Ze!